src/Giraffe/EndpointRouting.fs

```diff
@@ -0,0 +1,33 @@
+[<AutoOpen>]
+module Giraffe.EndpointRouting
+
+open System.Threading.Tasks
+open Microsoft.AspNetCore.Http
+open FSharp.Control.Tasks.V2
+
+open Microsoft.AspNetCore.Routing
+
+let private convertToHttpFunc (handler : HttpHandler) =
+    handler (Some >> Task.FromResult)
+
+let private convertToRequestDelegate (func : HttpFunc) =
+    fun (ctx : HttpContext) ->
+        task {
+            let! result = func ctx
+            match result with
+            | None   -> () // Shall we return here a HTTP 422 Unprocessable Entity?
+            | Some _ -> ()
+            return ()
+        } :> Task
+
+type IRouteBuilder with
+    member private this.MapGiraffeGetRoute (path : string, func : HttpFunc) =
+        let requestDelegate = convertToRequestDelegate func
+        this.MapGet(path, requestDelegate)
+        |> ignore
+
+    member this.MapGiraffe (routes : (string * HttpHandler) list) (handler : HttpHandler) =
```

Comment:
What's the trailing handler for?

Reply:
What, which trailing handler are you speaking of? *quicklyremovingandpushingnewchange*

It's not required, just a result of me typing before thinking.